modules/builtin/src/backend/variables/date.ts

```diff
@@ -15,6 +15,26 @@ class BoxedDate extends BaseVariable<BoxedDateType, DateConfig> {
     super(args)
   }
 
+  parse(text: string): BoxedDateType {
+    return moment(text).toDate()
+  }
+
+  equals({ other }: { other: Date }) {
+    return (
+      moment(this.value)
+        .toDate()
+        .getTime() === other.getTime()
+    )
+  }
+
+  isBefore({ other }: { other: Date }) {
+    return moment(this.value).toDate() < other
```

Comment:
this.value should already be a date, you should cast the other value instead

Reply:
The other value was already parsed by the `parse` function